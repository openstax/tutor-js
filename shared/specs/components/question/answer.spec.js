import Factories from '../../factories';
import Snapshot from 'react-test-renderer';
import Answer from '../../../src/components/question/answer';


// eslint-disable-next-line react/prop-types
jest.mock('../../../src/components/html', () => ({ className, html }) =>
    html ? <div className={className} dangerouslySetInnerHTML={{ __html: html }} /> : null
);

describe('Answer Component', function() {
    let props;
    let answer;

    beforeEach(() => {
        const exercise = Factories.exercise();
        answer = exercise.questions[0].answers[0];
        props = {
            answer,
            type: 'student',
            iter: 1,
            qid: answer.id,
            hasCorrectAnswer: false,
            onChangeAnswer: jest.fn(),
        };
    });

    it('renders answer', () => {
        const a = mount(<Answer {...props} />);
        expect(a).toHaveRendered('div.openstax-answer');
        expect(Snapshot.create(<Answer {...props} />).toJSON()).toMatchSnapshot();
    });

    it('renders answer feedback based on props', () => {
        props.show_all_feedback = true;
        const a = mount(<Answer {...props} />);
        expect(a).toHaveRendered('div.question-feedback-content');
        expect(Snapshot.create(<Answer {...props} />).toJSON()).toMatchSnapshot();
    });

    it('marks an answer as incorrect', () => {
        props.incorrectAnswerId = answer.id;
        props.hasCorrectAnswer = true;
        const a = mount(<Answer {...props} />);
        expect(a).toHaveRendered('.answer-incorrect');
        expect(Snapshot.create(<Answer {...props} />).toJSON()).toMatchSnapshot();
    });

});
