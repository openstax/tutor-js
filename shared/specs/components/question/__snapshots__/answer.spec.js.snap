// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer Component renders answer 1`] = `
<div
  className="openstax-answer"
>
  <section
    className="answers-answer"
    role="region"
  >
    <input
      checked={false}
      className="answer-input-box"
      disabled={false}
      id="1-option-1"
      name="1-options"
      onChange={[Function]}
      type="radio"
    />
    <label
      className="answer-label"
      htmlFor="1-option-1"
      onKeyPress={[Function]}
    >
      <span
        className="answer-letter-wrapper"
      >
        <button
          aria-label="Choice b:"
          className="answer-letter"
          data-test-id="answer-choice-b"
          disabled={false}
          onClick={[Function]}
        >
          b
        </button>
      </span>
      <div
        className="answer-answer"
      >
        <div
          className="answer-content"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<ol>
  <li>matter is moving at speeds of less than roughly 1% the speed of light,</li>
  <li>objects are too small to be seen with the naked eye, and</li>
  <li>there is the involvement of only a weak gravitational field</li>
</ol>
",
            }
          }
        />
      </div>
    </label>
  </section>
</div>
`;

exports[`Answer Component renders answer feedback based on props 1`] = `
<div
  className="openstax-answer"
>
  <section
    className="answers-answer"
    role="region"
  >
    <input
      checked={false}
      className="answer-input-box"
      disabled={false}
      id="3-option-1"
      name="3-options"
      onChange={[Function]}
      type="radio"
    />
    <label
      className="answer-label"
      htmlFor="3-option-1"
      onKeyPress={[Function]}
    >
      <span
        className="answer-letter-wrapper"
      >
        <button
          aria-label="Choice b:"
          className="answer-letter"
          data-test-id="answer-choice-b"
          disabled={false}
          onClick={[Function]}
        >
          b
        </button>
      </span>
      <div
        className="answer-answer"
      >
        <div
          className="answer-content"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Physics helps in predicting how burning fossil fuel releases pollutants.",
            }
          }
        />
        <aside>
          <div
            className="question-feedback-content has-html"
            dangerouslySetInnerHTML={
              Object {
                "__html": "This would be chemistry rather than physics.",
              }
            }
          />
        </aside>
      </div>
    </label>
  </section>
</div>
`;
